src/promptflow/promptflow/executor/flow_executor.py

```diff
@@ -1000,202 +965,10 @@ def apply_inputs_mapping(
-
-    @staticmethod
-    def _apply_inputs_mapping_for_all_lines(
-        input_dict: Mapping[str, List[Mapping[str, Any]]],
-        inputs_mapping: Mapping[str, str],
-    ) -> List[Dict[str, Any]]:
-        """Apply input mapping to all input lines.
-
-        For example:
-        input_dict = {
-            'data': [{'question': 'q1', 'answer': 'ans1'}, {'question': 'q2', 'answer': 'ans2'}],
-            'baseline': [{'answer': 'baseline_ans1'}, {'answer': 'baseline_ans2'}],
-            'output': [{'answer': 'output_ans1', 'line_number': 0}, {'answer': 'output_ans2', 'line_number': 1}],
-        }
-        inputs_mapping: {
-            "question": "${data.question}",  # Question from the data
-            "groundtruth": "${data.answer}",  # Answer from the data
-            "baseline": "${baseline.answer}",  # Answer from the baseline
-            "deployment_name": "text-davinci-003",  # literal value
-            "answer": "${output.answer}",  # Answer from the output
-            "line_number": "${output.line_number}",  # Answer from the output
-        }
+        # TODO: This function will be removed after the batch engine refactoring is completed.
+        from promptflow.batch._batch_inputs_processor import BatchInputsProcessor
 
-        Returns:
-        [{
-            "question": "q1",
-            "groundtruth": "ans1",
-            "baseline": "baseline_ans1",
-            "answer": "output_ans1",
-            "deployment_name": "text-davinci-003",
-            "line_number": 0,
-        },
-        {
-            "question": "q2",
-            "groundtruth": "ans2",
-            "baseline": "baseline_ans2",
-            "answer": "output_ans2",
-            "deployment_name": "text-davinci-003",
-            "line_number": 1,
-        }]
-        """
-        if inputs_mapping is None:
-            # This exception should not happen since developers need to use _default_inputs_mapping for None input.
-            # So, this exception is one system error.
-            raise UnexpectedError(
-                message_format=(
-                    "The input for batch run is incorrect. Please make sure to set up a proper input mapping before "
-                    "proceeding. If you need additional help, feel free to contact support for further assistance."
-                )
-            )
-        merged_list = FlowExecutor._merge_input_dicts_by_line(input_dict)
-        if len(merged_list) == 0:
-            raise InputMappingError(
-                message_format=(
-                    "The input for batch run is incorrect. Could not find one complete line on the provided input. "
-                    "Please ensure that you supply data on the same line to resolve this issue."
-                )
-            )
-
-        result = [FlowExecutor.apply_inputs_mapping(item, inputs_mapping) for item in merged_list]
-        return result
+        return BatchInputsProcessor.apply_inputs_mapping(inputs, inputs_mapping)
```

Comment:
Put this to a common util function, and import it in FlowExecutor, BatchInputsProcessor and promptflow._internal?

Reply:
Updated.